presets/ragengine/models.py

```diff
@@ -93,4 +103,26 @@ class QueryResponse(BaseModel):
 
 class HealthStatus(BaseModel):
     status: str
-    detail: Optional[str] = None 
+    detail: Optional[str] = None
+
+class ChatCompletionResponse(ChatCompletion):
+    source_nodes: Optional[List[NodeWithScore]] = None
+
+def messages_to_prompt(messages: List[Dict]) -> str:
+    """Convert messages to a prompt string."""
+    string_messages = []
+    for message in messages:
+        content = get_message_content(message)
+        string_messages.append(f"{message.get('role')}: {content}")
+    return "\n".join(string_messages)
+
+def get_message_content(message: Dict) -> str:
+    """Extract content from a ChatCompletionMessageParam."""
+    if message.get("role") == "user":
+        return message.get("content", {}).get("text", "") if isinstance(message.get("content"), dict) else message.get("content", "")
+    elif message.get("role") == "system":
+        return message.get("content", {}).get("text", "") if isinstance(message.get("content"), dict) else message.get("content", "")
+    elif message.get("role") == "developer":
+        return message.get("content", {}).get("text", "") if isinstance(message.get("content"), dict) else message.get("content", "")
```

Comment:
Any differences between these 3?

Reply:
updated to add user content handling and combine the rest